s3-proxy/nginx.conf

```diff
@@ -112,6 +112,60 @@ http {
             proxy_hide_header X-Archive-Files;
         }
 
+        location ~ ^/browse/.* {
+            proxy_pass $registry;
+            proxy_http_version 1.1;
+            proxy_buffering off;
+            proxy_ssl_verify on;
+            proxy_ssl_verify_depth 5;
+            proxy_ssl_trusted_certificate /etc/nginx/certs.pem;
+            proxy_ssl_name $REGISTRY_HOST;
+
+            proxy_set_header Host $REGISTRY_HOST;
+
+            # proxy_set_header Connection "";
+            # proxy_cache_bypass $http_pragma;
+            # proxy_redirect off;
+
+            # proxy_set_header X-Real-URL $upstream_http_location;
+            # resolver 8.8.8.8;
+
+            # add_header 'Access-Control-Allow-Origin' '*';
+            #
+            # proxy_set_header 'Content-Type' 'text/html';
+            # add_header 'Content-Type' 'text/html';
+
+            proxy_intercept_errors on;
+            error_page 301 302 307 = @handle_redirect;
+
+            # if ($upstream_http_location) {
+            #     proxy_pass $upstream_http_location;
+            # }
+        }
+
+        location @handle_redirect {
+            internal;
+            # TODO: this must be forced only for html files or, better yet,
+            # not forced at all -- we need to provide proper content-type headers
+            # when signing the urls in the registry
+            add_header 'Content-Type' 'text/html';
+
+            # Add CORS headers.
+            add_header 'Access-Control-Allow-Headers' $http_access_control_request_headers always;
+            add_header 'Access-Control-Allow-Methods' $http_access_control_request_method always;
+            add_header 'Access-Control-Allow-Origin' '*' always;
+            add_header 'Access-Control-Max-Age' '3000' always;
+            # add_header 'Access-Control-Expose-Headers' 'Content-Length, Content-Range, ETag, x-amz-meta-helium, x-amz-bucket-region, x-amz-delete-marker, x-amz-request-id, x-amz-version-id, x-amz-storage-class' always;
+
+            # Return success on OPTIONS.
+            if ($request_method = 'OPTIONS') {
+                return 200;
+            }
```

Comment:
This is only needed if you use XHR from JavaScript. If it's only used in an iframe, you don't need this at all - just the `proxy_pass`.

Reply:
well, js in iframe cold use xhr to fetch the adjacent data files, but according to the docs, preflight requests arent made in most cases we care about, so ima drop this